flow/client_server/static/internal/groups/group/js/script.js

```diff
@@ -0,0 +1,181 @@
+Vue.component('queue-row', {
+	props: ['name', 'inQueueSince', 'position'],
+	mounted: function() {
+		var self = this;
+
+        setInterval(function() {
+            self.$data.timeSinceInQueue = moment.utc(self.inQueueSince).toNow(true) + ' ago';
+        }, 1000);
+	},
+	data: function() {
+		return {
+			timeSinceInQueue: 'just now'
+		}
+	},
+	template:
+	'<tr>' +
+		'<td>{{ name }}</td>' +
+		'<td>{{ timeSinceInQueue }}</td>' +
+		'<td>#{{ position }}</td>' +
+	'</tr>'
+});
+
+Vue.component('queue', {
+	mounted: function() {
+		var self = this;
+		ws = new WebSocket(urls.websocket_server_address);
+
+		ws.onopen = function () {
+			ws.send(JSON.stringify({
+				aid: getAID()
+			}));
+		};
+
+		ws.onmessage = function (message) {
+			try {
+				var json = JSON.parse(message.data);
+				if (json.request === 'authentication update') {
+					ws.send(JSON.stringify({
+						request: 'switch_active_group',
+						group_id: $.QueryString.group_id
+					}))
+				} else if (json.request === 'switch active group confirmation') {
+					successAlert('Successfully Loaded Group', 1500)
+				} else if (json.request === 'queue update') {
+					console.log(json.in_queue);
+					self.$data.in_queue = json.in_queue;
+				} else {
+					console.log(json);
+				}
+
+			} catch (e) {
+				console.log('This doesn\'t look like a valid JSON: ', message.data);
+			}
+
+		};
+
+		ws.onclose = function () {
+			errorAlert('Oh no our connection closed!', 1500);
+		};
+
+		$('#join-queue').click(function() {
+			ws.send(JSON.stringify({
+				request: 'join_queue'
+			}));
+			successAlert('Joined the Queue', 1500);
+		});
+
+		var queueToggle = $('#queueToggle');
+
+		queueToggle.click(function() {
+			var text = queueToggle.text();
+			if (text === 'Leave Queue') {
+				ws.send(JSON.stringify({
+					request: 'leave_queue'
+				}));
+				queueToggle.removeClass('in-queue');
+				queueToggle.addClass('out-of-queue');
+				queueToggle.text('Join Queue');
+			} else if (text === 'Join Queue') {
```

Comment:
Yeah they are the recommended way to check for equality in javascript because javascript is php levels of heckery.

http://stackoverflow.com/questions/11234575/triple-3-equal-signs